src/rust/engine/engine_cffi/src/lib.rs

```diff
@@ -399,53 +399,105 @@ fn make_core(
   )
 }
 
-fn workunits_to_py_tuple_value<'a>(workunits: impl Iterator<Item = &'a WorkUnit>) -> Value {
-  let workunit_values = workunits
-    .map(|workunit: &WorkUnit| {
-      let mut workunit_zipkin_trace_info = vec![
-        (
-          externs::store_utf8("name"),
-          externs::store_utf8(&workunit.name),
-        ),
-        (
-          externs::store_utf8("start_secs"),
-          externs::store_u64(workunit.time_span.start.secs),
-        ),
-        (
-          externs::store_utf8("start_nanos"),
-          externs::store_u64(u64::from(workunit.time_span.start.nanos)),
-        ),
-        (
-          externs::store_utf8("duration_secs"),
-          externs::store_u64(workunit.time_span.duration.secs),
-        ),
-        (
-          externs::store_utf8("duration_nanos"),
-          externs::store_u64(u64::from(workunit.time_span.duration.nanos)),
-        ),
-        (
-          externs::store_utf8("span_id"),
-          externs::store_utf8(&workunit.span_id),
-        ),
-      ];
-      if let Some(parent_id) = &workunit.parent_id {
-        workunit_zipkin_trace_info.push((
-          externs::store_utf8("parent_id"),
-          externs::store_utf8(parent_id),
-        ));
-      }
+fn started_workunit_to_py_value(started_workunit: &StartedWorkUnit) -> Value {
+  use std::time::UNIX_EPOCH;
+  let duration = started_workunit
+    .start_time
+    .duration_since(UNIX_EPOCH)
+    .unwrap_or_else(|_| Duration::default());
+  let mut dict_entries = vec![
+    (
+      externs::store_utf8("name"),
+      externs::store_utf8(&started_workunit.name),
+    ),
+    (
+      externs::store_utf8("start_secs"),
+      externs::store_u64(duration.as_secs()),
+    ),
+    (
+      externs::store_utf8("start_nanos"),
+      externs::store_u64(duration.subsec_nanos() as u64),
+    ),
+    (
+      externs::store_utf8("span_id"),
+      externs::store_utf8(&started_workunit.span_id),
+    ),
+  ];
+
+  if let Some(parent_id) = &started_workunit.parent_id {
+    dict_entries.push((
+      externs::store_utf8("parent_id"),
+      externs::store_utf8(parent_id),
+    ));
+  }
 
-      if let Some(desc) = &workunit.metadata.desc.as_ref() {
-        workunit_zipkin_trace_info.push((
-          externs::store_utf8("description"),
-          externs::store_utf8(desc),
-        ));
-      }
+  if let Some(desc) = &started_workunit.metadata.desc.as_ref() {
+    dict_entries.push((
+      externs::store_utf8("description"),
+      externs::store_utf8(desc),
+    ));
+  }
 
-      externs::store_dict(&workunit_zipkin_trace_info.as_slice())
-    })
+  externs::store_dict(&dict_entries.as_slice())
+}
+
+fn workunit_to_py_value(workunit: &WorkUnit) -> Value {
+  let mut dict_entries = vec![
```

Comment:
This looks like a lot of duplication. Can this be deduplicated with the function above? If so, is it worth it?

Reply:
It could possibly be unduplicated, but I"m not sure it's worth it. If I was gonna do it, I'd want to do it in a different PR.